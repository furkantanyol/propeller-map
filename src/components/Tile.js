import React from 'react'
import PropTypes from 'prop-types'
import { TILE_LENGTH } from '../util/constants'

export default function Tile({ src, colIndex, rowIndex }) {
  return (
    <img
      style={{
        position: 'absolute',
        left: `${colIndex > 0 ? colIndex * TILE_LENGTH : 0}px`,
        top: `${rowIndex > 0 ? rowIndex * TILE_LENGTH : 0}px`
      }}
      src={src}
      alt='tile'
    />
  )
}

Tile.propTypes = {
  src: PropTypes.string.isRequired,
  colIndex: PropTypes.number.isRequired,
  rowIndex: PropTypes.number.isRequired
}
