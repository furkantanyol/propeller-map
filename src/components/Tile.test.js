import React from 'react'
import Tile from './Tile'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Tile src={'https://via.placeholder.com/256'} colIndex={1} rowIndex={1} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
