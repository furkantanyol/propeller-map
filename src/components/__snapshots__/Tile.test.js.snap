// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="tile"
  src="https://via.placeholder.com/256"
  style={
    Object {
      "left": "256px",
      "position": "absolute",
      "top": "256px",
    }
  }
/>
`;
